src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultIEnumerableConstructibleConverter.cs

```diff
@@ -12,11 +12,8 @@ internal sealed class DefaultIEnumerableConstructibleConverter : JsonEnumerableC
     {
         public static ConcurrentDictionary<Type, JsonPropertyInfo> s_objectJsonProperties = new ConcurrentDictionary<Type, JsonPropertyInfo>();
 
-        public override IEnumerable CreateFromList(ref ReadStack state, IList sourceList, JsonSerializerOptions options)
+        public static JsonPropertyInfo GetElementJsonPropertyInfo(JsonClassInfo elementClassInfo, JsonSerializerOptions options)
```

Comment:
I know you didn't change it in this PR 🤔, but L22 should be using `TryGetValue` to check for contains and return the value in one operation.

Reply:
Maybe this is more of a question for @steveharter ... What does this method mean exactly? (I am late to the party) 

From what I can figure out:
- `JsonClassInfo`  represents the **type** of the collection element. 
- It looks like this is creating a property with the containing class set to `object`

Consider this non-blocking, I'll keep reading to find out why!

Reply:
> Maybe this is more of a question for @steveharter ... What does this method mean exactly? (I am late to the party)

I wrote this method actually. In this instance, `JsonClassInfo` does represent the type of the collection element.
`JsonClassInfo` has a member type `JsonPropertyInfo` which, among other things, has a generic type parameter (the element type in this case) and contains helpers to (de)serialize the type.
This method helps us retrieve the appropriate `JsonPropertyInfo` for the collection type.
I can explain in more detail offline if this is still unclear.